codec-http2/src/main/java/io/netty/handler/codec/http2/Http2CodecUtil.java

```diff
@@ -190,6 +197,92 @@ public static void writeFrameHeader(ByteBuf out, int payloadLength, byte type,
     }
 
     /**
+     * Provides the ability to associate the outcome of multiple {@link ChannelPromise}
+     * objects into a single {@link ChannelPromise} object.
+     */
+    static class SimpleChannelPromiseAggregator implements ChannelFutureListener {
+        private final ChannelHandlerContext ctx;
+        private final ChannelPromise promise;
+        private int expectedCount;
+        private int successfullCount;
+        private boolean done;
+
+        SimpleChannelPromiseAggregator(ChannelHandlerContext ctx, ChannelPromise promise) {
+            assert ctx != null && promise != null;
+            this.ctx = ctx;
+            this.promise = promise;
+        }
+
+        /**
+         * Allocate a new promise which will be used to aggregate the overall success of this promise aggregator.
+         * @return A new promise which will be aggregated.
+         * {@code null} if {@link #doneAllocatingPromises()} was previously called.
+         */
+        public ChannelPromise newPromise() {
+            if (!done) {
+                ++expectedCount;
+                return ctx.newPromise().addListener(this);
```

Comment:
Yes I think this is a good idea.  I am looking into extending `DefaultChannelPromise` instead of directly implementing `ChannelPromise` to avoid duplication.